### PR TITLE
Enable proxy settings for docker run

### PR DESCRIPTION
**Pull request by qiluo-msft**
Signed-off-by: Qi Luo <qiluo-msft@users.noreply.github.com>

**- What I did**

**- How I did it**

**- How to verify it**
1. docker build
2. docker run -e "http_proxy=ip1" -e "ftp_proxy=ip2" -it <image>
3. start sshd inside container
4. ssh to localhost
5. verify 'env' has these proxy env vars

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by lguohan** [comment]
ENV seems to design for this purpose.

you can use docker run --env <key>=<value> to setup the env.

**Comment by qiluo-msft** [comment]
I tried ENV in experiments. It works in:
1. docker build time
2. attach to docker container

It doesn't work in: ssh to the daemon process running inside docker

This case is important since Jenkins master will ssh to the slave container and run all the job there.

File: dockers/docker-sonic-mgmt/Dockerfile.j2 [hunk 1]
```diff
@@ -96,3 +96,14 @@ RUN tar -xvzf azure-cli_bundle.tar.gz
...
+
+# Forward all proxy settings from 'docker run' command line to container scope startup script.
+# So even ssh client has them. All env variables are optional
+# Sample command:
+#   docker run -e "http_proxy=ip1" -e "ftp_proxy=ip2" -it <image>
+#
+ENTRYPOINT bash -c "touch /etc/profile.d/proxy.sh \
+    && echo 'export http_proxy=${http_proxy}' >> /etc/profile.d/proxy.sh \
+    && echo 'export https_proxy=${http_proxy}' >> /etc/profile.d/proxy.sh \
+    && echo 'export ftp_proxy=${http_proxy}' >> /etc/profile.d/proxy.sh \
```
Comment (lguohan) [inline dockers/docker-sonic-mgmt/Dockerfile.j2]: ${ftp_proxy}

File: dockers/docker-sonic-mgmt/Dockerfile.j2 [hunk 2]
```diff
@@ -96,3 +96,14 @@ RUN tar -xvzf azure-cli_bundle.tar.gz
...
 RUN git clone https://github.com/Azure/sonic-mgmt
+
+# Forward all proxy settings from 'docker run' command line to container scope startup script.
+# So even ssh client has them. All env variables are optional
+# Sample command:
+#   docker run -e "http_proxy=ip1" -e "ftp_proxy=ip2" -it <image>
+#
+ENTRYPOINT bash -c "touch /etc/profile.d/proxy.sh \
+    && echo 'export http_proxy=${http_proxy}' >> /etc/profile.d/proxy.sh \
+    && echo 'export https_proxy=${http_proxy}' >> /etc/profile.d/proxy.sh \
```
Comment (lguohan) [inline dockers/docker-sonic-mgmt/Dockerfile.j2]: http_proxy -> https_proxy

**Comment by lguohan** [comment]
ok

**Pull request merged**